llvm/compiler-course/llvm-ir/guseynov_e_lab2/add_pass.cpp

```diff
@@ -0,0 +1,92 @@
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/Transforms/Utils/BasicBlockUtils.h"
+
+namespace {
+struct ReplaceAddPass : llvm::PassInfoMixin<ReplaceAddPass> {
+  llvm::PreservedAnalyses run(llvm::Function &func,
+                              llvm::FunctionAnalysisManager &) {
+    if (func.getName() == "add") {
+      return llvm::PreservedAnalyses::all();
+    }
+
+    llvm::Module *mod = func.getParent();
+    if (!mod)
+      return llvm::PreservedAnalyses::all();
+
+    llvm::Function *addFunc = mod->getFunction("add");
+    if (!addFunc) {
+      return llvm::PreservedAnalyses::all();
+    }
+
+    if (addFunc->arg_size() != 2 && addFunc->isVarArg()) {
```

Comment:
Did you mean this? Please, double check
```suggestion
    if (addFunc->arg_size() != 2 || addFunc->isVarArg()) {
```